main.go

```diff
@@ -1,56 +1,84 @@
 package main
 
 import (
+	"context"
 	"fmt"
-	"log"
-	"net/http"
+	"log/slog"
+	"os"
 	"time"
 
+	"github.com/Hukyl/genesis-kma-school-entry/mail"
+	mailCfg "github.com/Hukyl/genesis-kma-school-entry/mail/config"
 	"github.com/Hukyl/genesis-kma-school-entry/models"
+	modelsCfg "github.com/Hukyl/genesis-kma-school-entry/models/config"
 	"github.com/Hukyl/genesis-kma-school-entry/rate"
 	"github.com/Hukyl/genesis-kma-school-entry/server"
+	serverCfg "github.com/Hukyl/genesis-kma-school-entry/server/config"
 	"github.com/Hukyl/genesis-kma-school-entry/settings"
-	"github.com/Hukyl/genesis-kma-school-entry/utils"
-	"github.com/gin-gonic/gin"
 )
 
-func NotifyUsers() {
-	rate, err := rate.NewNBURateFetcher().FetchRate("USD", "UAH")
+const emailInterval = 24 * time.Hour
+
+func NotifyUsers(ctx context.Context, apiClient server.Client, mc *mail.Client) {
+	db := apiClient.DB
+
+	rate, err := apiClient.RateFetcher.FetchRate("USD", "UAH")
 	if err != nil {
-		log.Println(err)
+		slog.Warn(err.Error())
 		return
 	}
-	db := models.NewDB()
 	var users []models.User
 	db.Connection().Find(&users)
-	log.Printf("Notifying by email %d users\n", len(users))
+	slog.Info(fmt.Sprintf(
+		"Notifying by email %d users\n",
+		len(users),
+	))
 	for _, user := range users {
 		message := fmt.Sprintf("1 USD = %f UAH", rate.Rate)
-		if err := utils.SendEmail(user.Email, message); err != nil {
-			log.Println(err)
+		if err := mc.SendEmail(ctx, user.Email, message); err != nil {
+			slog.Error(err.Error())
 		}
 	}
 }
 
 func main() {
-	settings.InitSettings()
-	models.NewDB().Migrate()
+	err := settings.InitSettings()
+	if err != nil {
```

Comment:
OPTIONAL: by the way, why don't you put this (and other similar cases) on the same line?
```
if err := settings.InitSettings(); err != nil {
}
```
I think it's looking good, not too long